evmbin/src/info.rs

```diff
@@ -102,13 +102,21 @@ pub fn run_action<T: Informant>(
 
 /// Execute given Transaction and verify resulting state root.
 pub fn run_transaction<T: Informant>(
+	// Chain specification name associated with the transaction
```

Comment:
Should these be doc-comments? I'm unsure of how the docs are rendered for private fields of a public struct.

Reply:
Thanks. I'm looking into it. If I update main.rs so that info.rs is public: `pub mod info;` then it's shown in the rustdocs, but if I make this a doc comment (i.e. `/// Chain ...`) and then build the docs with `cargo doc -p evmbin --release --open --no-deps` i get the following error.

```
error: documentation comments cannot be applied to method arguments
   --> evmbin/./src/info.rs:105:2
    |
105 |     /// Chain specification name associated with the transaction
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ doc comments are not allowed here
```

Reply:
@dvdplm I tried to make the input parameters into this `run_transaction` into a struct called `InputData`, since we can use doc comments on structs. The approach that I tried has been pushed in this commit https://github.com/paritytech/parity-ethereum/commit/8a6e6445e125ff1b0a60062da9327a55d3457667 (a separate branch to try it out), however I'm not sure how to overcome the `^^ undeclared lifetime
` error. In the `InputData` struct i'm not sure how to declare the lifetime parameter in addition to the generic type `T` (i.e. change `pub struct InputData<T> {` to `pub struct InputData<'a, T> {` or something?), and the existing code `mut informant: T,` is giving the following error and i'm not sure how to overcome that either:
```
122 |     mut informant: T,
    |     ^^^ expected identifier, found keyword
help: you can escape reserved keywords to use them as identifiers
    |
122 |     r#mut informant: T,
```

Reply:
See https://github.com/paritytech/parity-ethereum/commit/8a6e6445e125ff1b0a60062da9327a55d3457667#r33959733 – but I admit I'm not sure of the context here and why we need lifetimes on these values (prolly me being ignorant).

Reply:
Thanks for the tips! I somehow managed to get it to compile and I've created a separate PR https://github.com/paritytech/parity-ethereum/pull/10769. However there's quite a bit of code duplication and I haven't figured out how to approach resolving yet.